### DISCUSSION TITLE
Would love to have Last FM scobbling support.

### DISCUSSION THREAD

**Discussion by tanaybhomia** [MissingCore/Music#115] [Ideas]
The app is great works fine, Just somethings are missing like Search which I am sure are coming. But can we add the Last FM support ? It would be great.

**Comment by cyanChill** [comment]
Search is coming in the next major update, if you're using the APK version, you can try it out early with the [`v2.0.0-rc.1`](https://github.com/MissingCore/Music/releases/tag/v2.0.0-rc.1) release.

In regards to Last.fm support, there's no plans for it. Cost, complexity, and privacy are some of the reasons on why we won't implement this.
- Implementing this feature will require a server (which costs money) due to needing an API key (which should be kept private) to utilize their services (storing anything sensitive on the device to do this locally is considered not secured as reversed engineering can be used to obtain it).
- As we'll need a server, we'll also need authentication to prevent abuse and racking up unnecessary costs (also ensure those accessing the routes on the server are users of this app).
- If we need authentication, we need to deal with regulations and stuff as we're going to be dealing with personally identifiable information (PII), which I don't want to deal with.

**Comment by tanaybhomia** [comment]
Yeah , Thats fair. Its okay, App is great.

**Comment by tanaybhomia** [comment]
I tried the newest build for the app and I am really impressed, because the app also had some issues where it would not sort the albums correctly according to the meta data , some songs would be out of order. Just one thing we should improve the now playing screen , The previous now playing screen was actually good looking maybe because of the sound control which is added because and the playing bar of the song is actually blown apart and does not follow the same padding. Should I create a new discussion for this ?

**Reply by cyanChill** [reply]
With the issue with out of order tracks in albums, we now also consider disc number when sorting tracks within the album and fixed an issue where some metadata in `.flac` files weren't found (which included track number & disc number).

With the UI difference, a couple of things changes since then:
- Track name and artist name are now in a marquee (so the whole name should be displayed).
- The artwork is now just the artwork instead of artwork + vinyl. We plan in a future version after the major release to add an alternative version where it's an improved vinyl/disc look (the original design was replaced as it didn't really look good in light mode and kind of felt out of place).
- We've added the volume slider.
- We've added some additional quick actions/buttons for favoriting the track and viewing the upcoming tracks.

The padding between the seekbar and the volume bar relative to the sides of the device are the same (it might look different because we have the volume icons at both ends of the volume bar and slightly reduced the height of the bar as well). I guess one different relating to padding compared to the previous design is that we reduced the padding between the seekbar and the playback actions (shuffle/prev/play/pause/next/repeat) which might be throwing you off.

There are some potential changes that we can do:
- Increase the padding between the seekbar and the playback actions.
- Increase the vertical padding on the playback actions to give more space between it and the bars above and below it.

**Comment by tanaybhomia** [comment]
Yes , these changes might actually make the now playing screen better because that is just one thing that I didnt like in the app also the font size is slightly low as compared to the other app, I used the Native Photos gallery app on Nothing , the titles their are bit bigger and the overall font size is bit bigger, and would love to have the option to remove items which are not used on the bottom bar like Folders etc.

**Reply by cyanChill** [reply]
With the removal of some of the navigation in the bottom bar, that won't necessarily be possible. Though, in some future update, I do plan on enabling the ability to reorder the home screen tabs (excluding the primary "Home" screen), so you can place tabs such as "Folders" at the end.

**Comment by tanaybhomia** [comment]
Sure that would work, great update btw improves the app a lot. Just some visual touches to the now playing would be cherry on the top

**Comment by cyanChill** [comment]
The updated vinyl design and ability to reorder & hide the tabs after the "Home" tab has been added in the [`v2.1.0-rc.2`](https://github.com/MissingCore/Music/releases/tag/v2.1.0-rc.2) release (APK only).
